### PR TITLE
Deployment scripts updates

### PR DESCRIPTION
**Pull request by ben-kaufman**
Made some changes and added a new deployment script for full deployment of the contracts.

### CODE REVIEW

File: scripts/hatvaults-deploy.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  await hatVaults.deployed();
+
+  const hatVaultsImplementation = await getImplementationAddress(ethers.provider, hatVaults.address);
+  if (!silent) {
+
+    console.log("rewardController address:", rewardController.address);
+    console.log("hatVaults address:", hatVaults.address);
+    console.log("hatVaultsImplementation address:", hatVaultsImplementation);
+
+    if (network.name !== "hardhat") {
```
Comment (lirona) [inline scripts/hatvaults-deploy.js:95]: is this also supposed to happen only if not silent?

File: scripts/hattoken-deploy.js [hunk 2]
```diff
@@ -33,34 +32,21 @@ async function main() {
 
   console.log("hatToken address:", hatToken.address);
```
Comment (lirona) [inline scripts/hattoken-deploy.js:33]: here no option for silent?

File: scripts/hatvaults-deploy.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    "8210000000000000000000",
+    "7240000000000000000000",
+    "6390000000000000000000",
+  ],
+  epochLength = "195200",
+  governance = ADDRESSES[network.name].governance,
+  swapToken = ADDRESSES[network.name].hatToken,
+  whitelistedRouters = ADDRESSES[network.name].whitelistedRouters,
+  tokenLockFactory = ADDRESSES[network.name].tokenLockFactory,
+  silent = false
```
Comment (lirona) [inline scripts/hatvaults-deploy.js:39]: why only hat vaults has silent option?

File: scripts/hatvaults-deploy.js (hunk 1 above)
Comment (ben-kaufman) [reply scripts/hatvaults-deploy.js:95]: The silent is only good for using the script in tests, which is on hardhat anyway

File: scripts/hatvaults-deploy.js (hunk 3 above)
Comment (ben-kaufman) [reply scripts/hatvaults-deploy.js:39]: The silent is only good for using the script in tests and only that script is used in the tests currently

File: scripts/timelock-deploy.js [hunk 4]
```diff
@@ -34,15 +36,33 @@ async function main(config) {
...
       hatGovernanceDelay, // minDelay
       [governance], // proposers
       executors // executors
     );
-    await hatTimelockController.deployed();
+    await timelock.deployed();
+
+    ADDRESSES[network.name]["timelock"] = timelock.address;
+    fs.writeFileSync(__dirname + '/addresses.json', JSON.stringify(ADDRESSES, null, 2));
+    //verify
```
[the remark this answers is not in the document]
Comment (ben-kaufman) [reply scripts/timelock-deploy.js:50]: I think just delete that script

File: scripts/hatvaults-deploy.js (hunk 1 above)
Comment (lirona) [reply scripts/hatvaults-deploy.js:95]: i meant about this `if` clause, if it's supposed be under the `!silent` check

**Pull request closed without merging**